plugins/glustershd/actor.go

```diff
@@ -19,14 +21,52 @@ const (
 
 type shdActor struct{}
 
-func (actor *shdActor) Do(v *volume.Volinfo, key string, value string, logger log.FieldLogger) error {
-	if key != selfHealKey {
+func (actor *shdActor) Do(v *volume.Volinfo, key string, value string, opt xlator.VolumeOpType, logger log.FieldLogger) error {
+	voltype := v.Type.String()
+
+	if voltype != "Replicate" && voltype != "Disperse" {
 		return nil
 	}
+
 	glustershDaemon, err := newGlustershd()
 	if err != nil {
 		return err
 	}
+	if opt == xlator.VolumeStart {
```

Comment:
I'd like to see explicit switch case for each of the four operations to make it more readable. Currently, it is implicitly implied that if it's not start or stop, rest of the code is for set or reset

Reply:
done